manifests/init.pp

```diff
@@ -18,12 +18,30 @@
 # @param networkd_ensure
 #   The state that the ``networkd`` service should be in
 #
+# @param manage_timesyncd
+#   Manage the systemd tiemsyncd daemon
+#
+# @param timesyncd_ensure
+#   The state that the ``timesyncd`` service should be in
+#
+# @param $ntp_server
+#   comma separated list of ntp servers, will be combined with interface specific
+#   addresses from systemd-networkd. requires puppetlabs-inifile
+#
+# @param fallback_ntp_server
+#   A space-separated list of NTP server host names or IP addresses to be used
+#   as the fallback NTP servers. Any per-interface NTP servers obtained from
+#   systemd-networkd take precedence over this setting. requires puppetlabs-inifile
 class systemd (
-  Optional[Systemd::ServiceLimits] $service_limits  = undef,
-  Boolean                          $manage_resolved = false,
-  Enum['stopped','running']        $resolved_ensure = 'running',
-  Boolean                          $manage_networkd = false,
-  Enum['stopped','running']        $networkd_ensure = 'running',
+  Optional[Systemd::ServiceLimits] $service_limits   = undef,
+  Boolean                          $manage_resolved  = false,
+  Enum['stopped','running']        $resolved_ensure  = 'running',
+  Boolean                          $manage_networkd  = false,
+  Enum['stopped','running']        $networkd_ensure  = 'running',
+  Boolean                          $manage_timesyncd = false,
+  Enum['stopped','running']        $timesyncd_ensure = 'running',
+  Optional[String] $ntp_server                       = undef,
```

Comment:
Why not an array?

Reply:
I thougt about it. Systemd expects a comma separated list. What do you think about an array which I join()?

Reply:
yep, I think that's better.  The API you expose through puppet should make sense in puppet land.  Use join in the template or wherever.

Reply:
I think that would be a cleaner interface, personally (having it be structured data in Puppet, and join it later). Not sure whether it should require an array for a single element list, or whether it should allow you to pass in a string and force to list context.

Reply:
If you're being really friendly, I suppose you might also accept a string with no commas in instead of forcing people to use arrays of size 1.

Reply:
And if it's an array, the params should probably be `ntp_servers` and `fallback_ntp_server`?
This also makes it clear that the params can take multiple values

Reply:
@wyardley +1 Agreed.  If you expect an array, make the pluralise the parameter name.

Reply:
I would like to keep ntp_server and fallback_ntp_serve because those are the keys in the ini file. I updated the code to accept a single string or arrays. What do you think about that?

Reply:
I personally think it should be plural, will defer to @alexjfisher